Add meeting minutes to GitHub Pages

ianyong opened the pull request:
Changes:
- Site structure
  - Add `minutes` layout to Jekyll.
  - Add index page for meeting minutes that lists all pages using the `minutes` layout automatically.
  - Add link to the meeting minutes index page in the site header.
  - Update GitHub link in the header to point to this repository.
- Dependencies
  - Add `jekyll-spaceship` for greater control over site formatting.
    - Permission requested at nus-cs2103-AY2021S1/forum#267.
  - Remove `github-pages` gem.
    - So as to bypass safe mode and use non-whitelisted plugins.
  - Add previously packaged gems which were part of `github-pages` individually.
    - Disable `jekyll-github-metadata` as we are building the site files manually now.
- Deployment process
  - Add CI job to build the static files for the site before pushing to the orphaned `gh-pages` branch for deployment.
    - See https://github.com/nus-cs2103-AY2021S1/forum/issues/267#issuecomment-698113757 for the rationale behind this.
    - In order to get the generated static files to work with GitHub Pages' URL format, the `baseurl` has to be set as the repo name with a forward slash prepended:
      ```
      bundle exec jekyll build --baseurl /${{ github.event.repository.name }}
      ```
- Documentation
  - Migrate over all meeting minutes thus far.
    - 29 August 2020
    - 5 September 2020
    - 12 September 2020
    - 19 September 2020

To add a new meeting minutes, create a Markdown file in the `/docs/minutes` directory and append the following at the top of the file with the appropriate date substituted in the specified format:
```
---
layout: minutes
date: YYYY-MM-DD
---
```

**Note that for the date, leading zeroes should not be trimmed. Otherwise, sorting on the index page will not work.**

Resolves #17.

ianyong committed:
Add layout for minutes

ianyong committed:
Add automatic meeting minutes index page

ianyong committed:
Add meeting minutes for 29 August 2020

ianyong committed:
Add jekyll-spaceship

ianyong committed:
Update header links

ianyong committed:
Fix minutes formatting for 29 August 2020

ianyong committed:
Add meeting minutes for 5 September 2020

ianyong committed:
Add meeting minutes for 12 September 2020

ianyong committed:
Add meeting minutes for 19 September 2020

ianyong committed:
Remove trailing whitespace and add newline at EOF

ianyong committed:
Add CI job to build Jekyll without restrictions

The default GitHub Pages build process enforces safe mode which
disallows the use of any non-whitelisted dependencies.

wltan commented on .github/workflows/build-jekyll.yml:
```diff
@@ -0,0 +1,34 @@
+on:
+  push:
+    branches:
+      - master
```
Can we verify that this workflow runs smoothly on our repo?

Perhaps you can push an equivalent change directly to a non-master branch first, and see what happens to the `gh-pages` branch.

ianyong replied:
I thought that the `GITHUB_TOKEN` secret was being blocked due to this PR being from a fork. Seems to [also happen ](https://github.com/AY2021S1-CS2103T-W16-3/tp/runs/1159335401)with branches on this repo. :/

ianyong replied:
Apparently, `GITHUB_TOKEN` cannot be used; must be a GitHub user's personal access token.

ianyong replied:
CI job is able to run now. Tested on a branch of this repo as PRs from forks are unable to access secrets.

See https://github.com/AY2021S1-CS2103T-W16-3/tp/actions/runs/270453191.

wltan replied:
Ok, as seen from https://github.com/AY2021S1-CS2103T-W16-3/tp/commit/65e48f7a1f38c68e643cb29f8d29e0951f06bc4c the CI is able to build and deploy GitHub Pages properly.

ianyong commented:
While the current deployment process works with personal access tokens, @wltan brought up the point that we should have it work with the repository token `GITHUB_TOKEN` instead. This is a more ideal solution, especially if Prof Damith were to adopt the use of `jekyll-spaceship`.

ianyong committed:
Modify CI job to use 'GITHUB_TOKEN' instead

ianyong committed:
Add plugins used by GitHub pages

These plugins were previously packaged under the 'github-pages' gem
which had to be removed so as to remove the safe mode restrictions.

ianyong commented:
Relative links are currently not working. Trying to find a fix.

ianyong committed:
Disable jekyll-github-metadata

ianyong committed:
Add `baseurl` flag when building Jekyll

ianyong commented on .github/workflows/build-jekyll.yml:
```diff
@@ -0,0 +1,49 @@
+      - name: Build
+        env:
+          JEKYLL_ENV: production
+        run: bundle exec jekyll build --baseurl ${{ github.event.repository.name }} # Set the repo name as the base URL
```
Setting the `baseurl` to be the repo name is necessary for GitHub Pages to function.

ianyong committed:
Fix minutes index page not using base URL

ianyong committed:
Fix some links missing a '/'

This was achieved by prepending '/' before the repo name and using
the resulting string as the base URL.

zhaojj2209 reviewed (approved):
Tested and LGTM!

wltan reviewed (approved):
Commit https://github.com/AY2021S1-CS2103T-W16-3/tp/commit/8b4e933d0965311b61050847860770722963bb11 resulted in the creation of commit https://github.com/AY2021S1-CS2103T-W16-3/tp/commit/65e48f7a1f38c68e643cb29f8d29e0951f06bc4c on the `gh-pages` branch.

CI run: https://github.com/AY2021S1-CS2103T-W16-3/tp/runs/1160170726

LGTM!
